skin/frontend/base/default/js/oyst/oneclick.js

```diff
@@ -84,6 +84,22 @@ function oystOneClick(productTypeId, oneClickUrl, isProductAddtocartFormValidate
                         });
                     });
                     break;
+                case "virtual":
+                    // [Hook] Custom function allow anyone to use custom function to retrieve product id
```

Comment:
@rus-daniel all of this block is a copy/past, add virtual case after simple if there is no difference
````
case "simple":
case "virtual":